src/main/java/seedu/address/logic/commands/FilterCommand.java

```diff
@@ -0,0 +1,243 @@
+        }
+
+        FilterCommand otherFilterCommand = (FilterCommand) other;
+        return personFilter.equals(otherFilterCommand.personFilter);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("personFilter", personFilter)
+                .toString();
+    }
+
+    /**
+     * Stores the details to filter the contacts by. Contacts will be filtered by each non-empty field .
+     */
+    public static class PersonFilter {
+        private String name;
+        private String phone;
+        private String email;
+        private String address;
+        private Set<Tag> tags;
+
+        public PersonFilter() {}
+
+        /**
+         * Creates a copy of a filter with the same parameters as another filter.
+         *
+         * @param toCopy The other filter to create a copy of.
+         */
+        public PersonFilter(PersonFilter toCopy) {
+            setName(toCopy.name);
+            setPhone(toCopy.phone);
+            setEmail(toCopy.email);
+            setAddress(toCopy.address);
+            setTags(toCopy.tags);
+        }
+
+        /**
+         * Returns true if at least one field is edited.
+         */
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyNonNull(name, phone, email, address, tags);
+        }
+
+        public void setName(String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public void setPhone(String phone) {
+            this.phone = phone;
+        }
+
+        public String getPhone() {
+            return phone;
+        }
+
+        public void setEmail(String email) {
+            this.email = email;
+        }
+
+        public String getEmail() {
+            return email;
+        }
+
+        public void setAddress(String address) {
+            this.address = address;
+        }
+
+        public String getAddress() {
+            return address;
+        }
+
+        /**
+         * Sets {@code tags} to this object's {@code tags}.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public void setTags(Set<Tag> tags) {
+            this.tags = (tags != null) ? new HashSet<>(tags) : null;
+        }
+
+        /**
+         * Returns an unmodifiable tag set, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code tags} is null.
+         */
+        public Set<Tag> getTags() {
+            return (tags != null) ? Collections.unmodifiableSet(tags) : null;
+        }
+
+        /**
+         * Predicate to test if a person passes the filter by matching all parameters.
+         *
+         * @param person The person to be tested.
+         * @return a boolean representing whether the person passed the filter.
+         */
+        public boolean test(Person person) {
```

Comment:
Might be better to make it sound more like a boolean to clarify what test it is performing